DataAccessor/DataAccessor.cpp

```diff
@@ -0,0 +1,182 @@
+    if (!fs::exists(m_heaFilePath)) {
+        return false;
+    }
+
+    std::ifstream headerFile(m_heaFilePath);
+    for (std::string line; getline(headerFile, line);) {
+        if (line.at(0) == '#') { // The line starts with # is a comment
+            comment.push_back(line.substr(1));
+        }
+    }
+
+    char *fname = const_cast<char *>(m_heaFilePath.c_str());
+    char *record = const_cast<char *>(m_fileNameWithoutExtension.c_str());
+
+    int signalCount = 0;
+    // Find number of signals in specified record
+    for (int i = 0; i < 10; i++) {
+        signalCount = isigopen(record, fname, nullptr, i);
+        if (signalCount > 0) {
+            break;
+        }
+    }
+
+    if (signalCount < 1) {
+        return false;
+    }
+
+    m_signalCount = signalCount;
+    m_loaded = true;
+    parseComment();
+    return m_loaded;
+}
+
+bool DataAccessor::isLoaded() const {
+    return m_loaded;
+}
+
+int DataAccessor::signalCountGet() const {
+    if (isLoaded()) {
+        return m_signalCount;
+    }
+    return {};
+}
+
+std::vector<WFDB_Siginfo> DataAccessor::signalInfoGet() const {
+    if (isLoaded()) {
+        char *fname = const_cast<char *>(m_heaFilePath.c_str());
+        char *record = const_cast<char *>(m_fileNameWithoutExtension.c_str());
+        auto signalInfo = new WFDB_Siginfo[m_signalCount];
+        isigopen(record, fname, signalInfo, m_signalCount);
+        std::vector<WFDB_Siginfo> signals;
+        for (int i = 0; i < m_signalCount; i++) {
+            signals.push_back(signalInfo[i]);
+        }
+        delete[] signalInfo;
+        return std::move(signals);
+    }
+    return {};
+}
+
+std::vector<std::vector<float>> DataAccessor::signalDataGet() const {
+    if (isLoaded()) {
+        char *fname = const_cast<char *>(m_heaFilePath.c_str());
+        char *record = const_cast<char *>(m_fileNameWithoutExtension.c_str());
+
+        auto signalInfo = new WFDB_Siginfo[m_signalCount];
+        auto sample = new WFDB_Sample[m_signalCount];
+
+        // Open signal and save data in signalInfo
+        if (isigopen(record, fname, signalInfo, m_signalCount) == m_signalCount) {
+            std::vector<std::vector<float>> dataBuffer(m_signalCount);
+
+            std::vector<float> gain(m_signalCount);
+            std::vector<float> baseline(m_signalCount);
+
+            for (int i = 0; i < m_signalCount; i++) {
+                baseline[i] = static_cast<float>(signalInfo[i].baseline);
+                gain[i] = static_cast<float>(signalInfo[i].gain);
+            }
+
+            for (int i = 0; i < signalInfo->nsamp; i++) {
+                if (int readSampleError = getvec(sample) < 0) {
+                    std::cout << "error";
+                    exit(1);
+                }
+
+                for (int j = 0; j < m_signalCount; j++) {
+                    dataBuffer[j].push_back(((float) sample[j] - baseline[j]) / gain[j]);
```

Comment:
Jeżeli znamy ilość próbek dla poszczególnych sygnałów, proszę uwzględnij wywołanie `std::vector::reserve()` przed wypełnianiem wektora- spowoduje to zoptymalizowanie procesu wczytywania danych.